src/fileimpl.cpp

```diff
@@ -164,15 +164,35 @@ class TooSmallFile: public std::exception {};
   //////////////////////////////////////////////////////////////////////
   // FileImpl
   //
-  std::shared_ptr<FileImpl> FileImpl::BuildFileImpl(const std::string& filename) {
+  std::shared_ptr<FileImpl> FileImpl::BuildFileImpl(std::string fname) {
+    bool found = false;
+    std::shared_ptr<FileCompound> fileCompound;
     try {
-      return std::shared_ptr<FileImpl>(
-        new FileImpl(std::make_shared<FileCompound>(filename))
-      );
-    } catch (...) {
-      return std::shared_ptr<FileImpl>(
-        new FileImpl(std::make_shared<FileCompound>(filename, FileCompound::MultiPartToken::Multi))
-      );
+      fileCompound = std::make_shared<FileCompound>(fname);
+      found = true;
+    } catch(...) { }
+    if (found) {
+      try {
+        return std::shared_ptr<FileImpl>(new FileImpl(fileCompound));
+      } catch(TooSmallFile& e) {
+        if (fname.size() > 6 && fname.substr(fname.size()-6) == ".zimaa") {
+          // We have tried to open a split file. Let's continue with "aa" postfix removed.
+          fname.resize(fname.size()-2);
+        } else {
+          // We have found a (not split) file but it is too small.
+          throw ZimFileFormatError("Zim file(s) is too small.");
```

Comment:
"too small" → "of bad size or corrupted"

Reply:
Done